pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1536,32 +1550,47 @@ func SendProtobuf(url string, buf *bytes.Buffer, size int64,
 // Try all (first free, then rest) until it gets through.
 // Each iteration we try a different port for load spreading.
 // For each port we try all its local IP addresses until we get a success.
-func SendMetricsProtobuf(ctx *getconfigContext, ReportMetrics *metrics.ZMetricMsg,
-	iteration int) {
+func sendMetricsProtobufByURL(ctx *getconfigContext, metricsURL string,
+	ReportMetrics *metrics.ZMetricMsg, iteration int) {
+
 	data, err := proto.Marshal(ReportMetrics)
 	if err != nil {
-		log.Fatal("SendInfoProtobufStr proto marshaling error: ", err)
+		log.Fatal("sendMetricsProtobufByURL proto marshaling error: ", err)
 	}
 
 	buf := bytes.NewBuffer(data)
 	size := int64(proto.Size(ReportMetrics))
-	metricsUrl := zedcloud.URLPathString(serverNameAndPort, zedcloudCtx.V2API, devUUID, "metrics")
 	const bailOnHTTPErr = false
 	const withNetTrace = false
 	ctxWork, cancel := zedcloud.GetContextForAllIntfFunctions(zedcloudCtx)
 	defer cancel()
-	rv, err := zedcloud.SendOnAllIntf(ctxWork, zedcloudCtx, metricsUrl,
+	rv, err := zedcloud.SendOnAllIntf(ctxWork, zedcloudCtx, metricsURL,
 		size, buf, iteration, bailOnHTTPErr, withNetTrace)
 	if err != nil {
 		// Hopefully next timeout will be more successful
-		log.Errorf("SendMetricsProtobuf status %d failed: %s", rv.Status, err)
+		log.Errorf("sendMetricsProtobufByURL status %d failed: %s", rv.Status, err)
 		return
 	} else {
 		maybeUpdateMetricsTimer(ctx, true)
 		saveSentMetricsProtoMessage(data)
 	}
 }
 
+func sendMetricsProtobuf(ctx *getconfigContext,
```

Comment:
One of the implicit assumptions around the stillrunning logic for watchdogs is that each is that we only do one attempt to send before we call stillrunning again. This is because the http/tcp timeout is 2 minutes and the watchdog will trigger at 5 minutes so doing more than one send attempt is getting close to a spurious watchdog.

So I think the check for ctx.locConfig needs to be in the loop in metricsTimerTask.